libs/remix-ui/solidity-compiler/src/lib/contract-selection.tsx

```diff
@@ -186,6 +196,48 @@ export const ContractSelection = (props: ContractSelectionProps) => {
     return copyContractProperty('bytecode')
   }
 
+  /**
+   * Take AST and generates a UML diagram of compiled contract as svg
+   * @returns void
+   */
+  const generateUML = () => {
+    try {
+      const currentFile = api.currentFile
+      const ast = content4AST.length > 1 ? parser.parse(content4AST) : parser.parse(api.getCompilationResult().source.sources[currentFile].content)
+      const payload = vizRenderStringSync(convertUmlClasses2Dot(convertAST2UmlClasses(ast, currentFile)))
+      setSVGPayload(payload)
+      setShowViewer(!showViewer)
```

Comment:
could there be a PDF output, and save it in the filesystem?

Reply:
I am working on this in a second pr. I have been able to generate a pdf output but I am working on saving it to the our filesystem.

Reply:
Done